src/lab16.py

```diff
@@ -319,14 +319,11 @@ def layout(self, width, zoom):
             for obj in tree_to_list(self, []):
                 assert not obj.layout_needed()
 
-    def paint(self):
-        return []
-
     def paint_effects(self, cmds):
         if self.frame != self.frame.tab.root_frame and self.frame.scroll != 0:
             rect = skia.Rect.MakeLTRB(
-                self.x, self.y,
-                self.x + self.width, self.y + self.height)
+                self.x.get(), self.y.get(),
```

Comment:
Why the changes to add `.get()`?

Reply:
Yeah this is a good question. I am fairly confident the previous code was wrong (`DocumentLayout` no longer has a float `self.x` field) though I don't know how to trigger a crash in the old code (logically, scrolling inside a frame, but that doesn't seem to work?)

Reply:
Did you try a page with an iframe? That's what this code is for.

Reply:
Ok—I tested more and learned:
- Yes, this bug fix is necessary; scrolling in iframes in lab16 is broken on `main`
- There are other issues that have to be fixed, including a minor (no change in line count) fix to the book
- There's still a bug, wherein scrolling a frame doesn't seem to cause paint.